tests/test_edge_cases_comprehensive.py

```
@@ -0,0 +1,454 @@
+        from local_insight_engine.services.analysis_engine.claude_client import ClaudeClient
+
+        with patch('anthropic.Anthropic') as mock_anthropic:
+            # Mock timeout exception
+            mock_client = Mock()
+            mock_client.messages.create.side_effect = Exception("Request timeout")
+            mock_anthropic.return_value = mock_client
+
+            claude_client = ClaudeClient()
+
+            # Should handle timeout gracefully
+            with self.assertRaises(Exception):
```

Ruff (0.12.2)

Do not assert blind exception: `Exception`

(B017)